src/payment/strategies/chasepay/chasepay-payment-strategy.ts

```diff
@@ -0,0 +1,244 @@
+import { RequestSender, Response } from '@bigcommerce/request-sender';
+
+import {
+    PaymentActionCreator,
+    PaymentInitializeOptions,
+    PaymentMethod,
+    PaymentMethodActionCreator,
+    PaymentRequestOptions,
+    PaymentStrategyActionCreator,
+} from '../..';
+import { CheckoutActionCreator, CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { toFormUrlEncoded } from '../../../common/http-request';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { ChasePayScriptLoader, ChasePaySuccessPayload } from '../../../payment/strategies/chasepay';
+import { PaymentArgumentInvalidError } from '../../errors';
+import { CryptogramInstrument } from '../../payment';
+import PaymentStrategy from '../payment-strategy';
+import WepayRiskClient from '../wepay/wepay-risk-client';
+
+export default class ChasepayPaymentStrategy extends PaymentStrategy {
+    private _paymentMethod?: PaymentMethod;
+    private _logoContainer?: string;
+    private _button?: HTMLElement;
+
+    constructor(
+        store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _chasePayScriptLoader: ChasePayScriptLoader,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _requestSender: RequestSender,
+        private _wepayRiskClient: WepayRiskClient,
+        private _paymentStrategyActionCreator: PaymentStrategyActionCreator,
+        private _checkoutActionCreator: CheckoutActionCreator
+
+    ) {
+        super(store);
+    }
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
+
+        if (!options.chasepay) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.chasepayCheckoutOptions" argument is not provided.');
+        }
+
+        const {
+            onPaymentSelect = () => {},
+            logoContainer,
+            walletButton,
+        } = options.chasepay;
+
+        this._logoContainer = logoContainer;
+        const storeConfig = this._store.getState().config.getStoreConfig();
+
+        if (!storeConfig) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+        }
+
+        if (options.gatewayId === 'wepay') {
+            this._wepayRiskClient.initialize();
+        }
+
+        this._button = walletButton && document.getElementById(walletButton) || undefined;
+
+        if (this._button) {
+            this._button.addEventListener('click', () => this._handleWalletButtonClick(methodId));
```

Comment:
You need to retain a reference to the callback in order to remove the listener later. But here, you're passing an anonymous function therefore you're not able to remove the listener. That's why I suggested you to use the bind decorator.

Reply:
I use a function to call the `this. _handleWalletButtonClick` because the event listener expects an `EventListener` function, so i pass a anonymous function before calling the `_handleWalletButtonClick` one.

Reply:
What you wrote is incorrect. It doesn't actually remove the listener when you call `removeEventListener`. You need to change the signature of `_handleWalletButtonClick` so it becomes `EventListener` and pass it to `addEventListener`.